### PR TITLE
if we receive an array, use them as the ids to delete

### PR DESCRIPTION
**Pull request by gyula-s**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -369,19 +372,27 @@ Client.prototype.delete = function (query, options, finalCallback) {
     options = { commit: true }; // 'commit:true' option is default
   }
 
-  if (_.isString(query)) {
-    bodyQuery = query;
-  } else if (_.isObject(query)) {
-    bodyQuery = QueryString.stringify(query, " AND ", ":");
+  if (Array.isArray(query)) {
+    _.map();
```
Comment (jon-chard) [inline lib/client.js]: I guess this line isn't needed?

**Review by jon-chard — APPROVED** [review]
I don't like that the query parameter you pass in isn't always a query. But, whatever!

File: lib/client.js [hunk 2]
```diff
@@ -386,6 +386,34 @@ Client.prototype.delete = function (query, options, finalCallback) {
...
+ * @returns {Undefined|Promise}
+ */
+Client.prototype.deleteById = function (ids, options, finalCallback) {
+  var bodyData;
+  if (arguments.length < 3 && _.isFunction(options)) {
+    finalCallback = options;
+    options = { commit: true }; // 'commit:true' option is default
+  }
+
+  if (Array.isArray(ids)) {
```
Comment (jon-chard) [inline lib/client.js]: .. and if its not an array...? Can you delete one item by id?

Comment (gyula-s) [reply lib/client.js]: ideally... I'd expect anyone using this pass in an array...

Comment (jon-chard) [reply lib/client.js]: So what happens if they don't and we send an empty or undefined body to solr?

Comment (jon-chard) [reply lib/client.js]: You may as well put the post inside this check and return false or something if its not an array

**Review by jon-chard — APPROVED** [review]

**Pull request merged**